buildSrc/src/main/kotlin/com.qpay.java.integration-test.gradle.kts

```diff
@@ -0,0 +1,61 @@
+plugins {
+    java
+    idea
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_21
+}
+
+configurations {
+    compileOnly {
+        extendsFrom(configurations.annotationProcessor.get())
+    }
+}
+
+tasks.withType<Test> {
+    useJUnitPlatform()
+}
+
+val integrationTaskName = "integrationTest"
+
+sourceSets {
+    create(integrationTaskName) {
+        compileClasspath += sourceSets.main.get().output
+        runtimeClasspath += sourceSets.main.get().output
+    }
+}
+
+val integrationTest = task<Test>(integrationTaskName) {
+    description = "Runs integration tests."
+    group = "verification"
+
+    testClassesDirs = sourceSets[integrationTaskName].output.classesDirs
+    classpath = sourceSets[integrationTaskName].runtimeClasspath
+    shouldRunAfter("test")
+
+    useJUnitPlatform()
+}
+
+val integrationTestImplementation by configurations.getting {
+    extendsFrom(configurations.implementation.get())
+}
+val integrationTestRuntimeOnly by configurations.getting
+
+configurations["integrationTestRuntimeOnly"].extendsFrom(configurations.runtimeOnly.get())
+
+idea {
+    module {
+        testSources.from(sourceSets[integrationTaskName].java.srcDirs)
+    }
+}
+
+tasks.check { dependsOn(integrationTest) }
+
+dependencies {
+    integrationTestImplementation("org.junit.jupiter:junit-jupiter-api:${Versions.JUNIT_JUPITER_API}")
+    integrationTestImplementation("org.wiremock:wiremock-standalone:${Versions.WIREMOCK}")
+    integrationTestImplementation("org.springframework.boot:spring-boot-starter-test")
+    integrationTestRuntimeOnly("com.h2database:h2:${Versions.H2}")
```

Comment:
This is a specific dep for projects which use relation db. It cannot be common for all